.github/workflows/ci.yml

```diff
@@ -29,18 +29,19 @@ jobs:
       skip: ${{ steps.check_skip.outputs.SKIP }}
     steps:
       - name: Optimize CI
-        if: ${{ github.event_name == 'pull_request' }}
-        id: graphite-ci-action
-        uses: withgraphite/graphite-ci-action@main
-        with:
-          graphite_token: ${{ secrets.GRAPHITE_CI_OPTIMIZER_TOKEN }}
-      - name: Check if should skip the CI
         id: check_skip
         run: |
-          if [ "${{ steps.graphite-ci-action.outputs.skip }}" == "true" ]; then
-            echo "SKIP=true" >> $GITHUB_OUTPUT
-          else
+          BASE_BRANCH=${{ github.event.pull_request.base.ref }}
+          if [[ "$BASE_BRANCH" == "main" || "$BASE_BRANCH" =~ ^[0-9]{2}\.[0-9]{1,2}$ ]]; then
             echo "SKIP=false" >> $GITHUB_OUTPUT
+          else
+            response=$(curl -s -o /dev/null -w "%{http_code}" -X GET "${{ secrets.KVSTORE_URL }}/?key=base_${{ github.event.pull_request.head }}" \
+            -H "Authorization: Bearer ${{ secrets.KVSTORE_TOKEN }}")
+            if [ "$response" == "404" ]; then
+              echo "SKIP=false" >> $GITHUB_OUTPUT
+            else
+              echo "SKIP=true" >> $GITHUB_OUTPUT
+            fi
```

Comment:
Please check this part once.

Reply:
If something happens to kvstore, ci should still work fine.
That's why I changed it to default to setting skip to false and then changing it to true depending on the response to kvstore.